solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -138,82 +143,82 @@ private SolrZkClient(
       return;
     }
     this.zkServerAddress = zkServerAddress;
-    this.higherLevelIsClosed = higherLevelIsClosed;
-    if (strat == null) {
-      String connectionStrategy = System.getProperty("solr.zookeeper.connectionStrategy");
-      strat =
-          ZkClientConnectionStrategy.forName(connectionStrategy, new DefaultConnectionStrategy());
+    String chroot, zkHost;
+    int chrootIndex = zkServerAddress.indexOf('/');
+    if (chrootIndex == -1) {
+      zkHost = zkServerAddress;
+      chroot = null;
+    } else if (chrootIndex == zkServerAddress.length() - 1) {
+      zkHost = zkServerAddress.substring(0, zkServerAddress.length() - 1);
+      chroot = null;
+    } else {
+      zkHost = zkServerAddress.substring(0, chrootIndex);
+      chroot = zkServerAddress.substring(chrootIndex + 1);
     }
-    this.zkClientConnectionStrategy = strat;
 
-    this.solrClassLoader = solrClassLoader;
```

Comment:
I don't think we wanted to remove this line.

Reply:
Slightly concerning that that removal didn't cause some obvious test failure - but put it back in b8740424ceef85bf570c3d29774c2358cf382df9